packages/autocomplete-core/src/onKeyDown.ts

```diff
@@ -114,6 +114,14 @@ export function onKeyDown<TItem extends BaseItem>({
         .getState()
         .collections.every((collection) => collection.items.length === 0)
     ) {
+      // If requests are still pending when the panel closes, they could reopen
+      // the panel once they resolve.
+      // We want to prevent any subsequent query from reopening the panel
+      // because it would result in an unsolicited UI behavior.
```

Comment:
maybe here it should be qualified that the requests are still happening, but the handlers ignored, what do you think?

Reply:
Since we call `cancelAll` in multiple places, do you think it would make more sense to comment the method directly?

Reply:
I'll do this in a separate PR to keep this one focused.